velox/exec/tests/TableWriteTest.cpp

```diff
@@ -1101,6 +1101,110 @@ TEST_P(AllTableWriterTest, commitStrategies) {
   }
 }
 
+TEST_P(PartitionedTableWriterTest, specialPartitionName) {
+  int32_t numPartitions = 50;
+  int32_t numBatches = 2;
+
+  auto rowType =
+      ROW({"c0", "p0", "p1", "c1", "c3", "c5"},
+          {INTEGER(), INTEGER(), VARCHAR(), BIGINT(), REAL(), VARCHAR()});
+  std::vector<std::string> partitionKeys = {"p0", "p1"};
+  std::vector<TypePtr> partitionTypes = {INTEGER(), VARCHAR()};
+
+  std::vector charsToEscape = {
+      '"',
+      '#',
+      '%',
+      '\'',
+      '*',
+      '/',
+      ':',
+      '=',
+      '?',
+      '\\',
+      '\x7F',
+      '{',
+      '[',
+      ']',
+      '^'};
+  std::vector<RowVectorPtr> vectors = makeBatches(numBatches, [&](auto) {
```

Comment:
You can simplify by having a loop over different escape char? Then it might be easier to write the test. Thanks!